### PR TITLE
fix(rng): create rng only within isMainModule blocks

### PR DESCRIPTION
**Pull request by rymnc**
As a follow up to #1522, this method is more surgical, and does not make use of global variables (sort of).

Makes the rng required by all the procs which consume it, except for WakuNode.new(), because refactoring this will require updating a lot of tests, which can be done in a follow up PR.

### CODE REVIEW

File: apps/chat2/chat2.nim [hunk 1]
```diff
@@ -370,10 +370,10 @@ proc readInput(wfd: AsyncFD) {.thread, raises: [Defect, CatchableError].} =
     discard waitFor transp.write(line & "\r\n")
 
 {.pop.} # @TODO confutils.nim(775, 17) Error: can raise an unlisted exception: ref IOError
-proc processInput(rfd: AsyncFD) {.async.} =
+proc processInput(rfd: AsyncFD, rng: ref HmacDrbgContext) {.async.} =
```
Comment (rymnc) [inline apps/chat2/chat2.nim:373]: ideally this type would be aliased to something like `RngCtx`, but currently it does not justify a new file and import everywhere.

File: apps/wakunode2/config.nim [hunk 2]
```diff
@@ -53,7 +53,6 @@ type
 
     nodekey* {.
       desc: "P2P node private key as 64 char hex string.",
-      defaultValue: defaultPrivateKey()
       name: "nodekey" }: PrivateKey
```
Comment (LNSD) [inline apps/wakunode2/config.nim:56]: If we are not providing a default value, then let's use a `Option[PrivateKey]`:
```suggestion
      name: "nodekey" }: Option[PrivateKey]
```

File: apps/wakunode2/config.nim [hunk 3]
```diff
@@ -543,16 +539,19 @@ proc readValue*(r: var EnvvarReader, value: var crypto.PrivateKey) {.raises: [Se
...
-    let conf = WakuNodeConf.load(
+    var conf = WakuNodeConf.load(
       version=version,
       secondarySources = proc (conf: WakuNodeConf, sources: auto) =
         sources.addConfigFile(Envvar, InputFile("wakunode2"))
 
         if conf.configFile.isSome():
           sources.addConfigFile(Toml, conf.configFile.get())
     )
+    if $conf.nodekey == "":
```
Comment (LNSD) [inline apps/wakunode2/config.nim:552]: Then check if the value is `None`.

File: apps/wakunode2/wakunode2.nim [hunk 4]
```diff
@@ -215,7 +215,8 @@ proc retrieveDynamicBootstrapNodes*(dnsDiscovery: bool, dnsDiscoveryUrl: string,
 
 proc initNode(conf: WakuNodeConf,
               peerStore: Option[WakuPeerStorage],
-              dynamicBootstrapNodes: openArray[RemotePeerInfo] = @[]): SetupResult[WakuNode] =
+              dynamicBootstrapNodes: openArray[RemotePeerInfo] = @[],
+              rng: ref HmacDrbgContext): SetupResult[WakuNode] =
```
Comment (LNSD) [inline apps/wakunode2/wakunode2.nim:219]: As this parameter is mandatory, it should be located before the parameters with default values.

Doing that, you will avoid passing the RNG to this proc.

File: apps/wakunode2/config.nim [hunk 5]
```diff
@@ -543,16 +539,19 @@ proc readValue*(r: var EnvvarReader, value: var crypto.PrivateKey) {.raises: [Se
...
+      conf.nodekey = crypto.PrivateKey.random(Secp256k1, rng[]).tryGet()
```
Comment (LNSD) [inline apps/wakunode2/config.nim:553]: This default value assignment should be in the wakunode2 module, like the CLI argument validations.

**Review by LNSD — APPROVED** [review]
Please, check my comments 🙂

File: apps/wakunode2/config.nim (hunk 5 above)
Comment (rymnc) [reply apps/wakunode2/config.nim:553]: Do you mean, after reading the config, we check if nodekey is `none`, and then assign a random one? 

Something like this? 

```nim
let nodekey = if conf.nodekey.isSome(): conf.nodekey.get()
              else: crypto.PrivateKey.random(Secp256k1, rng[]).tryGet()
```

File: apps/wakunode2/config.nim (hunk 2 above)
Comment (rymnc) [reply apps/wakunode2/config.nim]: Addressed in 1b9566c8

File: apps/wakunode2/config.nim (hunk 3 above)
Comment (rymnc) [reply apps/wakunode2/config.nim]: Addressed in 1b9566c8

File: apps/wakunode2/wakunode2.nim (hunk 4 above)
Comment (rymnc) [reply apps/wakunode2/wakunode2.nim]: Addressed in 1b9566c8

File: apps/wakunode2/config.nim (hunk 5 above)
Comment (rymnc) [reply apps/wakunode2/config.nim]: Addressed in 1b9566c8. Please let me know if this is what you thought of.

Comment (rymnc) [reply apps/wakunode2/config.nim]: Thanks!

File: apps/chat2/chat2.nim (hunk 1 above)
Comment (LNSD) [reply apps/chat2/chat2.nim:373]: I thought the same 👍🏼

**Review by LNSD — APPROVED** [review]
LGTM, thanks ✅ 

P.S.: Note that there are conflicts after other PRs were merged.

**Pull request merged**